src/main/java/ch/njol/skript/expressions/ExprName.java

```diff
@@ -117,17 +118,16 @@ public class ExprName extends SimplePropertyExpression<Object, String> {
 
 	@Nullable
 	private static BungeeComponentSerializer serializer;
-	static final boolean HAS_GAMERULES;
 
 	static {
 		// Check for Adventure API
 		if (Skript.classExists("net.kyori.adventure.text.Component") &&
 				Skript.methodExists(Bukkit.class, "createInventory", InventoryHolder.class, int.class, Component.class))
 			serializer = BungeeComponentSerializer.get();
-		HAS_GAMERULES = Skript.classExists("org.bukkit.GameRule");
-		register(ExprName.class, String.class, "(1¦name[s]|2¦(display|nick|chat|custom)[ ]name[s])", "offlineplayers/entities/blocks/itemtypes/inventories/slots/worlds"
-			+ (HAS_GAMERULES ? "/gamerules" : ""));
-		register(ExprName.class, String.class, "(3¦(player|tab)[ ]list name[s])", "players");
+		register(ExprName.class, String.class, "(1:name[s])", "players/entities/inventories/named");
+		register(ExprName.class, String.class, "(2:(display|nick|chat|custom)[ ]name[s])", "players/entities/inventories/named");
+		register(ExprName.class, String.class, "(3:(player|tab)[ ]list name[s])", "players");
+		// we keep the entity input because we want to do something special with entities
```

Comment:
Is there still an AnyNamed converter for it? Might be worth having in case someone wants to use the type in another syntax

Reply:
Yes but this pattern has to take entity in preference here because it wants an `Entity` object rather than a converted `AnyNamed` object.